1.8/usage/service-discovery/haproxy-adminrouter.md

```diff
@@ -0,0 +1,107 @@
+---
+post_title: Configuring HAProxy in Front of Admin Router
+nav_title: HAProxy in Front of Admin Router
+menu_order: 400
+---
+
+You can use HAProxy to set up an HTTP proxy in front of the DC/OS [Admin Router](/docs/1.8/overview/concepts/#adminrouter).
+ 
+The HTTP Proxy must perform on-the-fly HTTP request and response header modification because DC/OS is not aware of the custom hostname and port that is being used by user agents to address the HTTP proxy.
+
+These instructions provide a tested [HAProxy](http://www.haproxy.org/) configuration example that handles the named request/response rewriting. This example ensures that the communication between HAProxy and DC/OS Admin Router is TLS-encrypted.
+
+
+1.  Install HAProxy [1.6.9](http://www.haproxy.org/#down).
+
+1.  Create an HAProxy configuration for DC/OS. This example is for a DC/OS cluster on AWS. For more information on HAProxy configuration parameters, see the [documentation](https://cbonte.github.io/haproxy-dconv/configuration-1.6.html#3).
+
+    ```
+    global
+      daemon
+      log 127.0.0.1 local0
+      log 127.0.0.1 local1 notice
+      maxconn 20000
+      pidfile /var/run/haproxy.pid
+    defaults
+      log            global
+      option         dontlog-normal
+      mode		 http
+      retries             3
+      maxconn          20000
+      timeout connect  5000
+      timeout client  50000
+      timeout server  50000
+    
+    frontend http
+      # Bind on port 9090. HAProxy will listen on port 9090 on each 
+      # available network for new HTTP connections.
+      bind 0.0.0.0:9090
```

Comment:
@joel-hamill @alberts Here I was missing how to specify the private key (which is required for configuring a server certificate) -- so I looked up the docs again, and it explains all the magic of the `crt` parameter here. I think we should add the documentation link here and slightly change wording. Can we change this block:
```
      # Specify a different server certificate.
      #
      # bind *:9091 ssl crt /path/to/browser-trusted.crt
      # Name of backend configuration for DC/OS.
```

to:
```
      # Specify your own server certificate chain and associated private key.
      # See https://cbonte.github.io/haproxy-dconv/configuration-1.6.html#5.1-crt
      # bind *:9091 ssl crt /path/to/browser-trusted.crt
      #
      # Name of backend configuration for DC/OS.
```